src/main/java/com/derongan/minecraft/deeperworld/synchronization/SectionSyncListener.kt

```diff
@@ -111,39 +103,77 @@ class SectionSyncListener : Listener {
+                        || !BlockLockerAPIv2.isAllowed(player, clicked, true)
+                        || !BlockLockerAPIv2.isAllowed(player, linkedBlock, true))
+                    return
+            }
 
-            if (!BlockLockerAPIv2.isAllowed(player, clicked, true) || !BlockLockerAPIv2.isAllowed(player, linkedBlock, true))
-                return
+            if (section.isOnTopOf(linkedSection)) return
 
             event.isCancelled = true
-            val otherInventory = ((linkedBlock.state as? Container) ?: return).inventory
-            val invList: List<ItemStack> = container.inventory.toList().filterNotNull()
-            if (invList.isNotEmpty()) {
-                //try adding items to the chest above, if something doesn't fit, drop it
-                invList.map { otherInventory.addItem(it).values }.flatten().also {
-                    if (it.isNotEmpty())
-                        player.sendMessage("${ChatColor.GOLD}This container had items in it, which have been ejected to synchronize it with the upper section.")
-                }.dropItems(loc, true)
-                container.inventory.clear()
+
+            //execute only if inventory successfully opened (e.x. not prevented by WorldGuard)
+            val linkedInventory = ((linkedBlock.state as? Container) ?: return).inventory
+            if (player.openInventory(linkedInventory) != null) {
+                logInfo("openeed successfully!")
+                //synchronize chests and drop anything that doesn't fit
+                val otherInventory = ((linkedBlock.state as? Container) ?: return).inventory
+                val invList: List<ItemStack> = container.inventory.toList().filterNotNull()
+                if (invList.isNotEmpty()) {
+                    //try adding items to the chest above, if something doesn't fit, drop it
+                    invList.map { otherInventory.addItem(it).values }.flatten().also {
+                        if (it.isNotEmpty())
+                            player.sendMessage("&6This container had items in it, which have been ejected to synchronize it with the upper section.".color())
+                    }.dropItems(loc, true)
+                    container.inventory.clear()
+                }
+
+                //add player to map of players using this inventory
+                if (!linkedBlock.chunk.isForceLoaded)
+                    keepLoadedInventories.getOrPut(linkedBlock.chunk, { mutableListOf() }) += player
             }
-            player.openInventory(((linkedBlock.state as? Container) ?: return).inventory)
         }
     }
 
-    /**
-     * Synchronize hopper pickups between sections
-     */
+    /** If there are any players accessing a synced inventory in this chunk, don't unload it*/
+    @EventHandler
+    fun onChunkUnload(cue: ChunkUnloadEvent) {
+        if (keepLoadedInventories.containsKey(cue.chunk)) cue.chunk.load()
```

Comment:
can we not just cancel?

Reply:
There's no cancel method on the event, and this seems to get the job done